format/Flight.proto

```diff
@@ -19,8 +19,17 @@
 syntax = "proto3";
 
 option java_package = "org.apache.arrow.flight.impl";
+option go_package = "ipc";
```

Comment:
Can you use `flight` namespace, there's already a component named `ipc` in the C++ code base and this could bring confusion.